### PR TITLE
DracoLoader: Fix not calling `onError()`.

### LINKED ISSUE #20883: Draco decoding errors result in uncaught promise rejections.
**Describe the bug**

When decoding a GLTF model with Draco compression, in some cases if the decoding process fails a promise in GLTFLoader is rejected but not caught. Thus in my code neither the onError nor the onLoad arguments are ever executed. 

I believe I have a fix here: https://github.com/mrdoob/three.js/compare/dev...twastvedt:dracoCatchPromise. Tested with the Draco example. I'm happy to create a PR, but I'm sure I'm ignorant of what all is at play here, so wanted to make sure first. My fix substitutes the current use of `decodeDracoFile`, which is marked deprecated in DRACOLoader, with `decodeGeometry`, which returns a promise that we can catch.

**To Reproduce**

1. Go to https://jsfiddle.net/srq463dg/
2. Note that in the console the onError method (line 58) never executes.
3. Open developer tools and note that an error has occurred in the Draco loader ("THREE.DRACOLoader: Decoding failed: Failed to decode point attributes").

Note: The `errorModel` data url in the fiddle is a valid gltf model that has been truncated. In my app's environment there is a string length limit imposed on models, thus this entire investigation.

***Code***

```js
loader.load(errorModel, function (data) {
    		console.log("Loaded model");
        scene.add(data.scene);
    }, null, function (error) {
    	console.log(`In error handler: ${error}`);
    });
```

**Expected behavior**

Any error in the Draco loader should cause the `onError` argument of `GLTFLoader.load` to be executed.

**Platform:**

 - Device: Desktop
 - OS: Windows
 - Browser: Chrome
 - Three.js version: dev

### PR DESCRIPTION
**Pull request by Benjythebee**
Fixed #20883

**Description**

I'm aware the dracoLoader might be getting a refactor sometime in the future, however in the meantime, this needed to be addressed;
The problem is that the onError is never called on failed DracoLoader load;

This PR fixes this issue by passing the promise reject to the onError argument

Please feel free to edit this PR if needed.

### CODE REVIEW

File: examples/jsm/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -1938,7 +1938,7 @@ class GLTFDracoMeshCompressionExtension {
 
 					resolve( geometry );
 
-				}, threeAttributeMap, attributeTypeMap );
+				}, threeAttributeMap, attributeTypeMap, undefined, reject );
```
Comment (Mugen87) [inline examples/jsm/loaders/GLTFLoader.js:1941]: How about we assign `LinearSRGBColorSpace` instead of `undefined`?

Comment (Benjythebee) [reply examples/jsm/loaders/GLTFLoader.js]: Brilliant, just pushed a commit to do just that! Thank you

File: examples/jsm/loaders/DRACOLoader.js [hunk 2]
```diff
@@ -82,13 +82,18 @@ class DRACOLoader extends Loader {
 
 	}
 
+
 	parse( buffer, onLoad, onError ) {
 
-		this.decodeDracoFile( buffer, onLoad, null, null, SRGBColorSpace ).catch( onError );
+		this.decodeDracoFile( buffer, onLoad, null, null, SRGBColorSpace ).catch( ( error ) => {
```
Comment (Mugen87) [inline examples/jsm/loaders/DRACOLoader.js:88]: Can we add a default callback parameter to `parse()` so we can simply write `catch( onError );`? Maybe:
```js
parse( buffer, onLoad, onError = () => {} ) {
```
My hope is that would clean the code a bit more up. Would this also work for `decodeDracoFile()`?

Comment (Benjythebee) [reply examples/jsm/loaders/DRACOLoader.js]: https://github.com/mrdoob/three.js/pull/27306/commits/14ddcc1779248e0983ea1a846dbfdee48fa25ee1

**Review by Mugen87 — APPROVED** [review]

File: examples/jsm/loaders/DRACOLoader.js (hunk 2 above)
Comment (Mugen87) [reply examples/jsm/loaders/DRACOLoader.js]: Awesome!

File: examples/jsm/loaders/DRACOLoader.js [hunk 3]
```diff
@@ -156,7 +157,12 @@ class DRACOLoader extends Loader {
 				} );
 
 			} )
-			.then( ( message ) => this._createGeometry( message.geometry ) );
+			.then( ( message ) => this._createGeometry( message.geometry ) )
+			.catch( ( error ) => {
+
+				throw new Error( error );
+
+			} );
```
Comment (donmccurdy) [inline examples/jsm/loaders/DRACOLoader.js:165]: nit: I think the catch-and-then-throw block is not really doing anything, won't we have the same behavior with or without it? If there's a special purpose for this we may need to call it out in a comment.

Comment (Mugen87) [reply examples/jsm/loaders/DRACOLoader.js:165]: Good point. Yes, I think the relevant addition is the new `catch()` when calling `decodeGeometry()`. This indeed seems rendundant.

@Benjythebee Can you confirm?

Comment (Benjythebee) [reply examples/jsm/loaders/DRACOLoader.js:165]: I apologize, I believe I did this for my own sanity while debugging this;
I can 100% send another PR to remove this portion of the code. Let me know!

Comment (Benjythebee) [reply examples/jsm/loaders/DRACOLoader.js:165]: https://github.com/mrdoob/three.js/pull/27308 feel free to mark as PR and merge; Close if not good

Comment (Mugen87) [reply examples/jsm/loaders/DRACOLoader.js:165]: Thanks! New PR already merged!

**Pull request merged**